modules/login/login_module.e

```diff
@@ -0,0 +1,1037 @@
+							end
+							create l_cookie.make ({LOGIN_CONSTANTS}.oauth_session, l_access_token.token)
+							l_cookie.set_max_age (l_access_token.expires_in)
+							l_cookie.set_path ("/")
+							res.add_cookie (l_cookie)
+						else
+
+							create {ARRAYED_LIST [CMS_USER_ROLE]}l_roles.make (1)
+							l_roles.force (l_user_api.authenticated_user_role)
+
+								-- Create a new user and oauth entry
+							create l_user.make (l_email)
+							l_user.set_email (l_email)
+							l_user.set_password (new_token) -- generate a random password.
+							l_user.set_roles (l_roles)
+							l_user.mark_active
+							l_user_api.new_user (l_user)
+
+								-- Add oauth entry
+							a_user_oauth_api.new_user_oauth2 (l_access_token.token, l_user_profile, l_user, "oauth_" + l_consumer.name )
+							create l_cookie.make ({LOGIN_CONSTANTS}.oauth_session, l_access_token.token)
+							l_cookie.set_max_age (l_access_token.expires_in)
+							l_cookie.set_path ("/")
+							res.add_cookie (l_cookie)
+							set_current_user (req, l_user)
+
+
+									-- Send Email
+							create es.make (create {LOGIN_EMAIL_SERVICE_PARAMETERS}.make (api))
+							write_debug_log (generator + ".handle register: send_contact_welcome_email")
+							es.send_contact_welcome_email (l_email, "")
+						end
+					else
+					end
+					r.set_redirection (req.absolute_script_url (""))
+					r.execute
+				end
+
+			end
+
+		end
+
+feature {NONE} -- Token Generation
+
+	new_token: STRING
+			-- Generate a new token activation token
+		local
+			l_token: STRING
+			l_security: SECURITY_PROVIDER
+			l_encode: URL_ENCODER
+		do
+			create l_security
+			l_token := l_security.token
+			create l_encode
+			from until l_token.same_string (l_encode.encoded_string (l_token)) loop
+				-- Loop ensure that we have a security token that does not contain characters that need encoding.
+			    -- We cannot simply to an encode-decode because the email sent to the user will contain an encoded token
+				-- but the user will need to use an unencoded token if activation has to be done manually.
+				l_token := l_security.token
+			end
+			Result := l_token
+		end
+
+
+
+feature {NONE} -- Implementation: date and time
+
+	http_date_format_to_date (s: READABLE_STRING_8): detachable DATE_TIME
+		local
+			d: HTTP_DATE
+		do
+			create d.make_from_string (s)
+			if not d.has_error then
+				Result := d.date_time
+			end
+		end
+
+	file_date (p: PATH): DATE_TIME
+		require
+			path_exists: (create {FILE_UTILITIES}).file_path_exists (p)
+		local
+			f: RAW_FILE
+		do
+			create f.make_with_path (p)
+			Result := timestamp_to_date (f.date)
+		end
+
+	timestamp_to_date (n: INTEGER): DATE_TIME
+		local
+			d: HTTP_DATE
+		do
+			create d.make_from_timestamp (n)
+			Result := d.date_time
+		end
+
+feature --{NONE} -- Helper OAUTH Consumers.
+
+
+	oauth_consumer_by_name (a_api: CMS_API; a_name: READABLE_STRING_8): detachable CMS_OAUTH_CONSUMER
```

Comment:
Don't you have a specific oauth module storage interface?
It would be cleaner than using SQL directly in that module class.